File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vow = 'aeiouAEIOU'
        cnt =0
        for r in range(k):
            if s[r] in vow:
                cnt +=1
        if len(s) >1: l=0
        else : return cnt
        maxx =cnt
        for r in range(k, len(s)): 
            if s[l] in vow: cnt-=1
            if s[r] in vow: cnt+=1
            if l< len(s)-k+1: l+=1
            maxx= max(maxx,cnt)
        return maxx


            

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int left=0, maxx=0, cnt=0;
        for(int rt=0; rt< s.size(); rt++){
            if(s[rt] == 'a' || s[rt] == 'e' || s[rt] == 'i' || s[rt] == 'o' || s[rt] == 'u') { 
                cnt++;
            }
            if(rt-left+1>k){
                if(s[left] == 'a' || s[left] == 'e' || s[left] == 'i' || s[left] == 'o' || s[left] == 'u') {
                    cnt--; 
                }
                left++;
            }

            maxx= max(maxx, cnt);
        }
        return maxx;
    }
};